editor/editor_interface.cpp

```diff
@@ -263,6 +265,77 @@ void EditorInterface::set_current_feature_profile(const String &p_profile_name)
 	EditorFeatureProfileManager::get_singleton()->set_current_profile(p_profile_name, true);
 }
 
+// Editor dialogs.
+
+void EditorInterface::popup_dialog_scene_tree(const Callable &p_callback, const TypedArray<StringName> &p_valid_types) {
+	if (dialog_scene_tree) {
+		dialog_scene_tree->queue_free();
+	}
+	dialog_scene_tree = memnew(SceneTreeDialog);
+	get_base_control()->add_child(dialog_scene_tree);
+
+	Vector<StringName> valid_types;
+	int length = p_valid_types.size();
+	valid_types.resize(length);
+	for (int i = 0; i < length; i++) {
+		valid_types.push_back(p_valid_types[i]);
+	}
+	dialog_scene_tree->set_valid_types(valid_types);
+
+	dialog_scene_tree->popup_scenetree_dialog();
+
+	Callable selected_callback = callable_mp(this, &EditorInterface::_node_selected).bind(p_callback);
+	dialog_scene_tree->connect(SNAME("selected"), selected_callback);
+
+	Callable canceled_callback = callable_mp(this, &EditorInterface::_node_selection_canceled).bind(p_callback);
+	dialog_scene_tree->connect(SNAME("canceled"), canceled_callback);
+}
+
+void EditorInterface::_node_selected(const NodePath &p_node_path, const Callable &p_callback) {
+	TypedArray<NodePath> node_paths;
+	NodePath path = get_edited_scene_root()->get_path().rel_path_to(p_node_path);
+	node_paths.push_back(path);
+	p_callback.call_deferred(node_paths);
+}
+
+void EditorInterface::_node_selection_canceled(const Callable &p_callback) {
+	p_callback.call_deferred(TypedArray<NodePath>());
+}
+
+void EditorInterface::popup_dialog_property_selector(Object *p_object, const Callable &p_callback, const Vector<int> &p_type_filter) {
+	if (dialog_property_selector) {
+		dialog_property_selector->queue_free();
+	}
+	dialog_property_selector = memnew(PropertySelector);
+	get_base_control()->add_child(dialog_property_selector);
+
+	Vector<Variant::Type> type_filter;
+	int length = p_type_filter.size();
+	type_filter.resize(length);
+	for (int i = 0; i < length; i++) {
+		type_filter.push_back((Variant::Type)p_type_filter[i]);
+	}
+	dialog_property_selector->set_type_filter(type_filter);
+
+	dialog_property_selector->select_property_from_instance(p_object);
+
+	Callable selected_callback = callable_mp(this, &EditorInterface::_property_selected).bind(p_callback);
+	dialog_property_selector->connect(SNAME("selected"), selected_callback);
+
+	Callable canceled_callback = callable_mp(this, &EditorInterface::_property_selection_canceled).bind(p_callback);
+	dialog_property_selector->connect(SNAME("canceled"), canceled_callback);
+}
+
+void EditorInterface::_property_selected(const String &p_property_name, const Callable &callback) {
+	TypedArray<NodePath> properties;
+	properties.push_back(NodePath(p_property_name).get_as_property_path());
+	callback.call_deferred(properties);
```

Comment:
Instead of `call_deferred()` you should connect this signal with `CONNECT_DEFERRED` and use `callp` for error reporting. Currently if you use wrong callback it just fails silently.